acme.go

```diff
@@ -186,6 +186,58 @@ func authorize(client *http.Client, config *Config, domain string) (ChallengeSet
 	return authzresp.ChallengeSet, nil
 }
 
+// Generates a key authorization string for a given token.
+func keyAuthorization(config *Config, token string) string {
+	return fmt.Sprintf("%s.%s", token, jwkThumbprint(&config.Key.PublicKey))
+}
+
+// acceptChallenge informs the server that the client accepts one of its challenges for validation.
+// The server will then perform the validation asynchronously.
+//
+// If client argument is nil, DefaultClient will be used.
+func acceptChallenge(client *http.Client, config *Config, challenge Challenge) (Challenge, error) {
```

Comment:
I wonder if we should switch to passing `Challenge` by reference, but not in this PR anyway.